test/src/tests/generated/body-complex/models.go

```diff
@@ -732,16 +857,69 @@ func (s *Sawshark) UnmarshalJSON(body []byte) error {
 	return nil
 }
 
-// Shark is
+// IShark
+type IShark interface {
+	AsSawshark() (*Sawshark, bool)
```

Comment:
Should IShark also implement IFish?

Reply:
you mean that IShark have all the `As*` methods as `IFish` ? I would say no since presumably you want to see the sub types.